### PR TITLE
[YANG]: Add Yang model support for Multi ASIC port and device_metadata

### PR DESCRIPTION
**Pull request by xincunli-sonic**
#### Why I did it

In Multi ASIC scenario, 
1. PORT configuration table requires `coreId`, `corePortId` and `numVoq`:
```json
{
    "PORT": {
        "Ethernet0": {
            "index": "0",
            "lanes": "101,102",
            "description": "fortyGigE1/1/1",
            "mtu": "9100",
            "alias": "fortyGigE1/1/1",
            "speed": "40000",
            "link_training": "off",
            "laser_freq": "191300",
            "tx_power": "-27.3",
            "dom_polling": "enabled",
            "coreId": "1",
            "corePortId": "1",
            "numVoq": "8"
        },
        "Ethernet1": {
            "index": "1",
            "lanes": "103,104",
            "description": "fortyGigE1/1/2",
            "mtu": "9100",
            "alias": "fortyGigE1/1/2",
            "admin_status": "up",
            "speed": "40000",
            "link_training": "on",
            "laser_freq": "191300",
            "tx_power": "-27.3",
            "dom_polling": "enabled",
            "coreId": "0",
            "corePortId": "14",
            "numVoq": "8"
        },
        "Ethernet63": {
            "index": "63",
            "lanes": "87,88",
            "description": "fortyGigE1/4/16",
            "mtu": "9100",
            "alias": "fortyGigE1/4/16",
            "speed": "40000",
            "laser_freq": "191300",
            "tx_power": "-27.3",
            "dom_polling": "disabled",
            "coreId": "0",
            "corePortId": "15",
            "numVoq": "8"
        }
    }
}
```
2. DEVICE_METADATA configuration table requires `asic_id`:
```json
{
"DEVICE_METADATA": {
        "localhost": {
        "asic_id": "06:00.0",
        "asic_name": "asic0",
        "hwsku": "Force10-S6100",
        "default_bgp_status": "up",
        "docker_routing_config_mode": "unified",
        "hostname": "sonic-s6100-01",
        "platform": "x86_64-dell_s6100_c2538-r0",
        "mac": "4c:76:25:f4:70:82",
        "default_pfcwd_status": "disable",
        "bgp_asn": "65100",
        "deployment_id": "1",
        "type": "ToRRouter",
        "bgp_adv_lo_prefix_as_128" : "true",
        "buffer_model": "traditional",
        "yang_config_validation": "disable",
        "rack_mgmt_map": "dummy_value",
        "timezome": "Europe/Kiev"
    }
  }
}

```

##### Work item tracking
- Microsoft ADO **(number only)**: 27252814, 27253157

#### How I did it
1. Added `coreId`, `corePortId` and `numVoq` field to CONFIG_DB PORT table.
1. Added `asic_id` and `asic_name` field to CONFIG_DB DEVICE_METADATA table.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-device_metadata.yang [hunk 1]
```diff
@@ -40,6 +40,14 @@ module sonic-device_metadata {
                     type stypes:hwsku;
                 }
 
+                leaf asic_id {
+                    type string {
+                        length 0..16;
+                    }
+                    description "On a VoQ switch, the ASIC Id is used as a identifier in global
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-device_metadata.yang:47]: nit: this asic_id is applicable for any multi asic device.

Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang:47]: `asic_id unique identifier of the asic used by SAI for initialization`

File: src/sonic-yang-models/yang-models/sonic-device_metadata.yang [hunk 2]
```diff
@@ -40,6 +40,13 @@ module sonic-device_metadata {
                     type stypes:hwsku;
                 }
 
+                leaf asic_id {
+                    type string {
+                        length 0..16;
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-device_metadata.yang:45]: >0 [](http://example.com/codeflow?start=31&length=1)

Is 0 valid?

Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-device_metadata.yang:45]: >16 [](http://example.com/codeflow?start=34&length=2)

where is 16 from?

Comment (xincunli-sonic) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang:45]: Yes, it is.

https://www.rfc-editor.org/rfc/rfc7950#section-9.4.4: 
> A length value is a non-negative integer or one
   of the special values "min" or "max".

Comment (xincunli-sonic) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang:45]: @arlakshm , can you help confirm that we can just put `type string` without length?

Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang:45]: No, we cannot have empty string

Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang]: I understand 1 now. How about 16?

Comment (xincunli-sonic) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang]: @arlakshm, any insight about the upper limit, its format was like `"asic_id": "06:00.0"` in configdb.

Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang]: this is the asic pcie address, so 16 is lowest number I can think. If this is concern we can increase to higher value

**Review by arlakshm — APPROVED** [review]
lgtm

**Comment by xincunli-sonic** [comment]
@gechiang would you mind backport to Chassis 202205 branch.

**Comment by gechiang** [comment]
@xincunli-sonic this PR failed to be able to cleanly picked into the msft repo.  Please raise a PR directly under the msft repo 202205 branch for this change and tag me for review.
Thanks!

**Pull request merged**